### DISCUSSION TITLE
`sqlx::query().bind()` is not binding the values in my query

### DISCUSSION THREAD

**Discussion by Sammyalhashe** [rwf2/Rocket#2695] [Questions] [answered]
I am trying to run a simple query: `sqlx::query("SELECT * FROM x LIMIT ?").bind(num)` and my postgres db is only getting the query with the `?` in it. Does anyone have any ideas?

**Comment by wezm** [comment, marked as answer]
In postgres you have to use $1, $2 etc for bind params. https://www.postgresql.org/docs/16/sql-prepare.html See also the note on [bind](https://docs.rs/sqlx/latest/sqlx/query/struct.Query.html#method.bind):

> bind parameters that appear in the query (? for most SQL flavors, $1 .. $N for Postgres)

**Reply by Sammyalhashe** [reply]
doh! I didn't think this through! Thanks for the answer :)
